org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/adapter/Adapters.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * SonarLint for Eclipse
+ * Copyright (C) 2015-2017 SonarSource SA
+ * sonarlint@sonarsource.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarlint.eclipse.core.internal.adapter;
+
+import javax.annotation.Nullable;
+import org.eclipse.core.runtime.AssertionFailedException;
+import org.eclipse.core.runtime.IAdaptable;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.core.runtime.PlatformObject;
+
+/**
+ * Copied from Oxygen for backward compatibility with older version of Eclipse
+ */
+public class Adapters {
+  /**
+   * If it is possible to adapt the given object to the given type, this
+   * returns the adapter. Performs the following checks:
+   * 
+   * <ol>
+   * <li>Returns <code>sourceObject</code> if it is an instance of the
+   * adapter type.</li>
+   * <li>If sourceObject implements IAdaptable, it is queried for adapters.</li>
+   * <li>Finally, the adapter manager is consulted for adapters</li>
+   * </ol>
+   * 
+   * Otherwise returns null.
+   * 
+   * @param sourceObject
+   *            object to adapt, can be null
+   * @param adapter
+   *            type to adapt to
+   * @param allowActivation
+   *            if true, plug-ins may be activated if necessary to provide the requested adapter.
+   *            if false, the method will return null if an adapter cannot be provided from activated plug-ins.
+   * @return a representation of sourceObject that is assignable to the
+   *         adapter type, or null if no such representation exists
+   */
+  @SuppressWarnings("unchecked")
+  public static <T> T adapt(@Nullable Object sourceObject, Class<T> adapter, boolean allowActivation) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this method 6, down to the maximum allowed 5. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1142)